### PR TITLE
coral-hive:  Added support for WITH clauses

### PR DESCRIPTION
**Pull request by uzshao**

### CODE REVIEW

**Comment by uzshao** [comment]
@wmoustafa @funcheetah @antumbde This is a follow-up diff to support WITH clauses in Hive.  Please take a look when you get a chance.

**Review by ljfgem — APPROVED** [review]
LGTM, thank you @uzshao 
Integration test succeeded.

File: coral-hive/src/test/java/com/linkedin/coral/hive/hive2rel/HiveToRelConverterTest.java [hunk 1]
```diff
@@ -62,6 +62,40 @@ public void testBasic() {
     verifyRel(rel, expected);
   }
 
+  @Test
+  public void testBasic2() {
```
Comment (wmoustafa) [inline coral-hive/src/test/java/com/linkedin/coral/hive/hive2rel/HiveToRelConverterTest.java:66]: Can we share the common code between `testBasic` and `testBasic2`?

Comment (wmoustafa) [reply coral-hive/src/test/java/com/linkedin/coral/hive/hive2rel/HiveToRelConverterTest.java:66]: @findepi @losipiuk FYI.

**Comment by wmoustafa** [comment]
@findepi @losipiuk FYI.

File: coral-hive/src/test/java/com/linkedin/coral/hive/hive2rel/HiveToRelConverterTest.java (hunk 1 above)
Comment (uzshao) [reply coral-hive/src/test/java/com/linkedin/coral/hive/hive2rel/HiveToRelConverterTest.java:69]: Just fixed it by extracting the common code into a separate function.

**Comment by uzshao** [comment]
@wmoustafa Can you take a look again?  I extracted the common code, and also added one more test case on with.

File: coral-hive/src/test/java/com/linkedin/coral/hive/hive2rel/HiveToRelConverterTest.java [hunk 2]
```diff
@@ -62,6 +60,49 @@ public void testBasic() {
...
+    RelNode rel = converter.convertSql(sql);
+    RelBuilder relBuilder = createRelBuilder();
+    RelNode expected = relBuilder.scan(ImmutableList.of("hive", "default", "foo"))
+        .project(ImmutableList.of(relBuilder.field("b"), relBuilder.field("c")), ImmutableList.of(), true).build();
+    verifyRel(rel, expected);
+  }
+
+  @Test
+  public void testWithNested() {
+    // Test if the code can handle the use of the second alias from the WithList
```
Comment (ljfgem) [inline coral-hive/src/test/java/com/linkedin/coral/hive/hive2rel/HiveToRelConverterTest.java:97]: nit: the comment is the same as `testWith2`, should also be changed?

Comment (uzshao) [reply coral-hive/src/test/java/com/linkedin/coral/hive/hive2rel/HiveToRelConverterTest.java]: Nice catch!  Just pushed a fix to the comment.

**Review by wmoustafa — APPROVED** [review]

**Pull request merged**